server.rb

```diff
@@ -5,33 +5,29 @@
 require_relative 'bot'
 require 'easy_translate'
 
-EasyTranslate.api_key= ENV['TRANSLATE_API_KEY']
-
-@@initial = 0
-@@users = {}
-
-def language(key)
-  if @@users[key] == "English"
-    {to: :en}
-  elsif @@users[key] == "Vietnamese"
-    {to: :vi}
-  elsif @@users[key] == "French"
-    {to: :fr}
-  else
-    {to: :en}
-  end
-end
-
 get '/callback' do
   if params["hub.mode"] == "subscribe" && params["hub.verify_token"] == "asdf123"
     params["hub.challenge"]
   else
     ap "Give me Parameters"
-    ap params
     "Unknown Parmeters"
   end
 end
 
+EasyTranslate.api_key= ENV['TRANSLATE_API_KEY']
+@@users = {}
+
+def language(id)
+  languages = EasyTranslate::LANGUAGES.invert
+  choice = @@users[id].downcase
+  languages.each do |key, value|
+    if key == choice
+      return {to: value.to_sym}
+    end
+  end
+  return nil
+end
```

Comment:
see my comment below about not having to return `{to: ...}` structure. that is not the concern of this method.

then this function is just one line like this: `EasyTranslate::LANGUAGES.invert[@@users[id].downcase])` (you don't need to convert `.to_sym` either because the API allows strings too.

i'm not saying you should use one line; i just mean you can make the logic cleaner :)

also, when `@@users[id]` is nil or when the language is not found, why don't you just return `en` for English?